### PR TITLE
SQL Datasources: Update behavior of default connection limits

### PR DESCRIPTION
**Pull request by codeincarnate**
This PR is a follow-up to #65834 to improve the behavior of defaults set in the `custom.ini` file. This PR exposes those settings to the front-end so that they'll be used when interacting with new datasources, and datasources that haven't set these values set.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by zoltanbedi — APPROVED** [review]
LGTM

File: public/app/features/plugins/sql/components/configuration/useMigrateDatabaseFields.test.ts [hunk 1]
```diff
@@ -57,8 +69,9 @@ describe('Database Field Migration', () => {
       ...defaultProps,
       onOptionsChange: (options: DataSourceSettings) => {
         const jsonData = options.jsonData as SQLOptions;
-        expect(jsonData.maxOpenConns).toBe(SQLConnectionDefaults.MAX_CONNS);
-        expect(jsonData.maxIdleConns).toBe(Math.ceil(SQLConnectionDefaults.MAX_CONNS));
+        expect(jsonData.maxOpenConns).toBe(10);
```
Comment (baldm0mma) [inline public/app/features/plugins/sql/components/configuration/useMigrateDatabaseFields.test.ts:72]: Nice

Comment (baldm0mma) [reply public/app/features/plugins/sql/components/configuration/useMigrateDatabaseFields.test.ts:72]: Love seeing updated tests. :)

**Review by baldm0mma — APPROVED** [review]
great work; lgtm.

**Pull request merged**